### PR TITLE
refactor(api-tests): #649 refactor tests api

### PR DESCRIPTION
**Pull request by ConradBoydElliottGustafson**

### CODE REVIEW

File: server/auth_function/requirements.txt
Comment (ConradBoydElliottGustafson) [inline server/auth_function/requirements.txt:1]: Satisfy renovate issue

File: docker-compose-testcontainer.yml
Comment (ConradBoydElliottGustafson) [inline docker-compose-testcontainer.yml:1]: Added another container with a dependency on the fam-flyway to get rid of the testcontainer race condition. The test fixture waits for nginx to come online, proving that fam-flyway is done

File: server/auth_function/test/conftest.py
Comment (ConradBoydElliottGustafson) [inline server/auth_function/test/conftest.py:1]: Got rid of the race condition
Switched to use the env variables from the local-dev.env

File: server/backend/api/app/database.py
Comment (ConradBoydElliottGustafson) [inline server/backend/api/app/database.py:1]: Got rid of test code from a production file

File: server/backend/api/config/config.py
Comment (ConradBoydElliottGustafson) [inline server/backend/api/config/config.py:1]: Got rid of test code from a production file

File: server/backend/local-dev.env
Comment (ConradBoydElliottGustafson) [inline server/backend/local-dev.env:1]: Added FC_API_TOKEN so the backend can startup. Still needs a real value to run tests.

File: server/backend/requirements.txt
Comment (ConradBoydElliottGustafson) [inline server/backend/requirements.txt:1]: Moved boto libraries to the top and pinned the version. There was a dependency conflict with another library and this fixes it.

File: server/backend/testspg/conftest.py
Comment (ConradBoydElliottGustafson) [inline server/backend/testspg/conftest.py:1]: Refactored and cleaned up test code

File: server/backend/testspg/crud/test_crud_application.py
Comment (ConradBoydElliottGustafson) [inline server/backend/testspg/crud/test_crud_application.py:1]: Got rid of a dependency on DB ordering of results that was failing tests if the results came back in the wrong order.
Reformatted

File: server/backend/testspg/router/test_router_application.py
Comment (ConradBoydElliottGustafson) [inline server/backend/testspg/router/test_router_application.py:1]: Got rid of a dependency on DB ordering of results that was failing tests if the results came back in the wrong order.

**Review by MCatherine1994 — APPROVED** [review]

**Pull request merged**